models/Translation/Listing/Dao.php

```diff
@@ -125,18 +125,24 @@ public function loadRaw()
      */
     public function load()
     {
-        //$allTranslations = $this->getAllTranslations();
-        $translations = [];
         $this->model->setGroupBy($this->getDatabaseTableName() . '.key', false);
 
         $queryBuilder = $this->getQueryBuilder([$this->getDatabaseTableName() . '.key']);
-        $translationsData = $this->db->fetchAllAssociative((string) $queryBuilder, $this->model->getConditionVariables());
+        $cacheKey = $this->getDatabaseTableName().'_data_' . md5((string)$queryBuilder);
 
-        foreach ($translationsData as $t) {
-            $transObj = Model\Translation::getByKey(id: $t['key'], domain: $this->model->getDomain(), languages: $this->model->getLanguages());
+        if (!empty($this->model->getConditionParams()) || !$translations = Cache::load($cacheKey)) {
```

Comment:
Should we also invalidate cache on translation save? the below use case fails with this change:

```php
        $translations = new Translation\Listing();
        $translations->setDomain('messages');
        $translations->load();

        //add new translation
        $translation = new Translation();
        $translation->setKey('test' . rand());
        $translation->setDomain('messages');
        $translation->save();

        $translations->load();  //load translations excluding new one 
```

Reply:
Yeah we should. I just checked with when adding a TK in the backend not in the code

Reply:
The caching also works with the api. Problem here is that you need to actually set translations in the api, otherwise it won't get even saved.